src/fsipy/automatedPostprocessing/log_plotter.py

```diff
@@ -0,0 +1,908 @@
+    parser.add_argument("--plot-cfl", action="store_true", help="Plot CFL numbers (mean, min and max)")
+    parser.add_argument("--plot-reynolds", action="store_true", help="Plot Reynolds numbers (mean, min and max)")
+    parser.add_argument("--probe-points", type=int, nargs="+", help="List of probe points to plot")
+    parser.add_argument("--start-cycle", type=int, default=1,
+                        help="Cycle to start plotting data from (default: 1, start at first cycle)")
+    parser.add_argument("--end-cycle", type=int, default=None,
+                        help="Cycle to end plotting data at (default: end at last cycle)")
+    parser.add_argument("--compute-average", action="store_true",
+                        help="Compute average over cycles. This option has no effect for " +
+                             "--plot-newton-iteration-atol and --plot-newton-iteration-rtol.")
+    parser.add_argument("--compare-cycles", action="store_true",
+                        help="Compare data across multiple cycles. This option has no effect for " +
+                             "--plot-newton-iteration-atol and --plot-newton-iteration-rtol.")
+    parser.add_argument("--figure-size", type=lambda x: tuple(map(int, x.split(','))), default=(10, 6),
+                        help="Figure size in inches (width, height), e.g., '12,8' (default: 10,6)")
+    parser.add_argument("--save", action="store_true", help="Save the figures to files")
+    parser.add_argument("--output-directory", type=str, default="Images",
+                        help="Directory where plot images will be saved (default: 'Images')")
+    parser.add_argument("--log-level", type=int, default=20,
+                        help="Specify the log level (default is 20, which is INFO)")
+
+    return parser.parse_args()
+
+
+def main() -> None:
+    args = parse_command_line_args()
+
+    # Enable --plot-all by default if no specific --plot options are provided
+    plot_types = ['cpu_time', 'ramp_factor', 'pressure', 'newton_iteration_atol', 'newton_iteration_rtol',
+                  'probe_points', 'flow_rate', 'velocity', 'cfl', 'reynolds']
+    args.plot_all = args.plot_all or all(not getattr(args, f'plot_{plot_type}') for plot_type in plot_types)
+
+    # Create logger and set log level
+    logging.basicConfig(level=args.log_level, format="%(message)s")
+
+    # Log a warning if --compute-average and --compare-cycles are used together
+    if args.compute_average and args.compare_cycles:
+        logging.warning("WARNING: Select either --compute-average or --compare-cycles, not both.")
+        args.compare_cycles = False
+
+    # Parse log data
+    parsed_dict = parse_dictionary_from_log(args.log_file)
+    parsed_data = parse_log_file(args.log_file)
+
+    # Extract end time, cycle length and time step size from the parsed dictionary
+    end_time = parsed_dict.get("T", 0.951)
+    cycle_length = parsed_dict.get("T_cycle", 0.951)
```

Comment:
If `T_cycle` is based on offset_stenosis problem, it should be `T_Cycle`? maybe I need to fix the offset_stenosis since using capital letter is a bit weird...

Reply:
Yes, it is based on the offset stenosis problem. In VaMPy, `cardiac_cycle` is used in `Artery.py` and `Atrium.py`. Should we also use that?

Reply:
Yes, `cardiac_cycle` seems better!

Reply:
Ok, I will update the script to use that instead.

Reply:
Thanks!